### DISCUSSION TITLE
[feature] implement the: auto_now_add auto_now

### DISCUSSION THREAD

**Discussion by zodman** [encode/orm#94] [Ideas]

**Comment by aminalaee** [comment]
@zodman thanks for this. Would you like to implement this?

**Comment by zodman** [comment]
Sorry man! It's something the project missing without it. I'm will jump to another project to use...

**Comment by aminalaee** [comment]
Thanks for that. I will try to implement that sometime later.

**Comment by Priyansh2001here** [comment]
I am looking forward to contribute to this project, I Guess starting with this would be a good start.
This will be my first open source contribution so, if there's another Issue (simpler than this) i would like to work on it

**Reply by aminalaee** [reply]
That's great. I think it should be good for start.
if you need any help, feel free to just ask here.

**Comment by Priyansh2001here** [comment]
I am trying to dig , but there's one thing I am unable to understand 

How does [this](https://github.com/encode/orm/blob/master/orm/models.py#L407) snippet work

```
        validator = typesystem.Schema(
            fields={key: value.validator for key, value in fields.items()}
        )

        print(kwargs)
        kwargs = validator.validate(kwargs)
```

validate method fills respected keys with default values

**Reply by aminalaee** [reply]
Hmm...to be honest I was thinking maybe your change should be more in this area: https://github.com/encode/orm/blob/master/orm/models.py#L412

And in that part, we check if field is readonly and has default (value or callable), then we assign that default to the kwargs.
If I'm not mistaken if we added `auto_now` the field becomes read_only in Django admin too. So we should make fields with these attributes read-only.

What do you think?

**Reply by aminalaee** [reply]
yeah, Looking at the Note, Django sets editable=False.
https://docs.djangoproject.com/en/3.2/ref/models/fields/#datefield

**Reply by Priyansh2001here** [reply]
> Hmm...to be honest I was thinking maybe your change should be more in this area: https://github.com/encode/orm/blob/master/orm/models.py#L412
> 
> And in that part, we check if field is readonly and has default (value or callable), then we assign that default to the kwargs. If I'm not mistaken if we added `auto_now` the field becomes read_only in Django admin too. So we should make fields with these attributes read-only.
> 
> What do you think?

I was just digging around code base 
I will try to add ```auto_now=True``` attribute just like ```default=``` and if there's auto now make that field read only and adding value as ```datetime.date.today()```

**Comment by Priyansh2001here** [comment]
having some trouble in implementing ```auto_now_add=True``` i.e automatically updating update time

**Reply by aminalaee** [reply]
What's the issue?
